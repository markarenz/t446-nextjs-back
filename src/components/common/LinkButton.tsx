import Link from 'next/link';
import styles from '../../styles/modules/Button.module.scss';

type Props = {
  href: string;
  children: JSX.Element;
  variant?: string;
  color?: string;
  size?: string;
  tabIndex?: number;
};

const LinkButton: React.FC<Props> = ({
  href,
  children,
  variant,
  color,
  size
}) => {
  const classSize = size === 'small' ? styles.small : '';
  const classVariant =
    variant === 'outlined' ? styles.outlined : styles.contained;
  let classColor = styles.colorPrimary;
  if (color === 'secondary') {
    classColor = styles.colorSecondary;
  }
  return (
    <Link href={href} passHref>
      <div
        className={`${styles.button} ${classSize} ${classVariant} ${classColor}`}
        data-testid="link-button-inner"
      >
        {children}
      </div>
    </Link>
  );
};
export default LinkButton;
