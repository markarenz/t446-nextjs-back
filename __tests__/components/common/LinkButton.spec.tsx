import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import LinkButton from '../../../src/components/common/LinkButton';

const defaultProps = {
  href: '/test'
};
const altProps = {
  href: '/test',
  variant: 'outlined',
  color: 'secondary',
  size: 'small'
};
describe('LinkButton', () => {
  it('renders link button', async () => {
    render(
      <LinkButton {...defaultProps}>
        <span>Test</span>
      </LinkButton>
    );
    const element = await screen.findByTestId('link-button-inner');
    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
  });
  it('renders link button', async () => {
    render(
      <LinkButton {...altProps}>
        <span>Test</span>
      </LinkButton>
    );
    const element = await screen.findByTestId('link-button-inner');
    expect(element).toBeInTheDocument();
  });
});
