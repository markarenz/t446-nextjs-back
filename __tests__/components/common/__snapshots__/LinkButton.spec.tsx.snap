// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders link button 1`] = `
<div
  class="button  contained colorPrimary"
  data-testid="link-button-inner"
  href="/test"
>
  <span>
    Test
  </span>
</div>
`;
